core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -243,6 +244,12 @@ public Builder setCurrentVersionId(int newVersionId) {
         changes.add(new MetadataUpdate.SetCurrentViewVersion(newVersionId));
       }
 
+      historyEntry =
```

Comment:
Instance field assignment should be prefixed with `this.` to be clear that it is modifying more than local state.